python/tests/transform_coords_test.py

```diff
@@ -0,0 +1,342 @@
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Unless I missed something, all tests use trees. It would be good to also check with non-tree-acyclic graphs, (Do they have a name?) e.g. a diamond: 
```
   a
 /   \
b     c
 \   /
   d
```

Reply:
Done.